File: SORTS/Insertionsort.c
#include<stdio.h>
int main()
{
    int a[20],n,i,j,k,l,m,temp;
    printf("**********This is Insertion sort program**********");

    printf("\nHow many element do you want to enter:");
    scanf("%d",&n);
    for(i=0; i<n; i++)
    {
        scanf("%d",&a[i]);
    }

    printf("'\nYour array elements are:");
    for(j=0; j<n; j++)
    {
        printf("%d",a[j]);
    }
   

    for(k=1; k<n; k++)
    {
        temp=a[k];
        l=k-1;
        while(l>=0 && a[l]>temp)
        {
            a[l+1]=a[l];
            l--;
        }

        a[l+1]=temp;
    }

     printf("'\nYour array elements are sorted:");
     for(m=0; m<n; m++)
    {
        printf("%d",a[m]);
    }

    return 0;

}

File: SORTS/InsertionSort.java
import java.util.Scanner;
public class InsertionSort {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int a [] = new int[20];
        int temp;

        System.out.print("How many element do you want to enter:");
        int n = sc.nextInt();
        for(int i=0; i<n; i++){
            a[i] = sc.nextInt();
        }

        System.out.println("Your array elements before sorting ");
        for(int i=0; i<n; i++){
            System.out.println(a[i]);
        }
        for(int i=1; i<n; i++){
            temp = a[i];
            int j = i-1;
            while(j>=0 && a[j]>temp){
                a[j+1] = a[j];
                j--;
            }
            a[j+1] = temp;
        }



        System.out.println("Your array elements after sorting ");
        for(int i=0; i<n; i++){
            System.out.println(a[i]);
        }
    }
}
